src/lib/ppx_version/lint_version_syntax.ml

```diff
@@ -493,20 +412,10 @@ let lint_ast =
                 if acc.in_functor then validate_neither_bin_io_nor_version
                 else validate_version_if_bin_io
           in
-          let versioned_type_misuse_errors_fun =
-            if not @@ in_versioned_type_module acc.module_path then
-              get_versioned_type_misuses
-            else if not acc.in_versioned_ext then not_in_versioned_ext_fun
```

Comment:
Even if you allow mentions of specific versions, don't you want to keep the lint warning when a versioned type is used without `%%verioned`? 

For example, this type declaration currently generates a warning:
```
module Foo = struct
  module Stable	= struct
    module V1 =	struct
      type t = int
    end
  end
end
```

Reply:
Ok. So then we still want to return `not_in_versioned_ext_fun` if `in_versioned_type_module acc.module_path && not acc.in_versioned_ext`?